### ISSUE TITLE
Extension `ext-hash` is not required when `hash` function is used

### ISSUE THREAD

**Issue by ruudk** [shipmonk-rnd/composer-dependency-analyser#204]
Not sure if I'm doing something wrong, but I noticed that this analyzer does not complain when using the `hash` function:
https://github.com/twigstan/twigstan/blob/2acea9a4cff92b158a68ed891dc1dc196b896c1d/src/Error/ErrorCollapser.php#L29-L32

without requiring the `hash` extension.
https://github.com/twigstan/twigstan/blob/2acea9a4cff92b158a68ed891dc1dc196b896c1d/composer.json#L9-L23

**Comment by JanTvrdik** [comment]
Probably relevant https://wiki.php.net/rfc/permanent_hash_ext

**Comment by janedbal** [comment]
Exactly, core extensions are never reported as those are always present.

**Comment by ruudk** [comment]
Hmm, interesting. 

That does mean that all these libraries declare it for no reason? 
https://packagist.org/packages/ext-hash/dependents?order_by=downloads

**Comment by janedbal** [comment]
It does. But since it does not hurt, we decided not to report such thing.

**Comment by janedbal** [comment]
But, it sometimes changes with PHP version, so those core ext-* requirements may be valid for old PHP versions (< PHP 7.4 for hash one)
